odl/tomo/geometry/spect_geometry.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2014 - 2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""SPECT geometry."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+# External
+import numpy as np
+
+# Internal
+import odl
+from odl.tomo.geometry.detector import Flat2dDetector
+from odl.tomo.geometry.geometry import (Geometry,
+                                        AxisOrientedGeometry)
+from odl.tomo.util.utility import perpendicular_vector
+
+__all__ = ('ParallelHoleCollimatorGeometry', )
+
+
+class ParallelHoleCollimatorGeometry(Geometry, AxisOrientedGeometry):
+    """Geometry for SPECT Parallel hole collimator."""
+
+    def __init__(self, apart, dpart, det_radius, axis=[0, 0, 1],
+                 **kwargs):
+        """Initialize a new instance.
+        Parameters
+        ----------
+        apart : 1-dim. `RectPartition`
+            Partition of the angle interval
+        dpart : 2-dim. `RectPartition`
+            Partition of the detector parameter rectangle
+        det_radius : `list` of positive `float`
+            A list of radii for every position of the detector
+            Individual radii must be positive.
+            If all radii are equal, `list` of one radius is sufficient,
+            otherwise a radius per angular position is needed.
```

Comment:
Ah, I think you misunderstood me. We should accept both a list and a single float, but *store* it internally always as a list. You can use the following code (assuming `length` is known):

```python
try:
    rad_list = [float(r) for r in det_rad]
    if len(rad_list) != length:
        raise ValueError('Expected {} entries in radius sequence, got {}.'
                         ''.format(length, len(rad_list)))
except TypeError:
    # Triggered when we have a single float
    rad_list = [float(det_rad)] * length
```